roles/sap_vm_provision/tasks/platform_ansible/kubevirt_vm/execute_main.yml

```diff
@@ -68,24 +61,14 @@
         - not lookup('ansible.builtin.vars', loop_item, default='') is skipped
         - lookup('ansible.builtin.vars', loop_item, default='') is failed
 
-- name: Write private ssh key to ansible_controller
-  delegate_to: "{{ sap_vm_provision_kubevirt_vm_ansible_controller }}"
-  no_log: true
-  ansible.builtin.copy:
-    dest: "{{ sap_vm_provision_ssh_host_private_key_file_path }}"
-    content: "{{ sap_vm_provision_ssh_host_private_key_data }}"
-    mode: "0600"
-  when: sap_vm_provision_kubevirt_vm_guest_ssh_auth_mechanism == "private_key_data"
-
 - name: Ansible Task block to execute on target inventory hosts
   remote_user: "{{ sap_vm_provision_kubevirt_vm_os_user }}"
   become: true
   become_user: root
   delegate_to: "{{ inventory_hostname }}"
   vars:
-    ansible_password: "{{ sap_vm_provision_kubevirt_vm_os_user_password }}"
     ansible_ssh_private_key_file: "{{ sap_vm_provision_ssh_host_private_key_file_path }}"
-    ansible_ssh_common_args: "-o ConnectTimeout=180 -o ControlMaster=auto -o ControlPersist=3600s -o UserKnownHostsFile=/dev/null -o ForwardX11=no -o ProxyJump={{ __sap_vm_provision_kubevirt_vm_register_execution_host_user }}@{{ sap_vm_provision_execution_host }}"
+    ansible_ssh_common_args: "-o StrictHostKeyChecking=no -o ConnectTimeout=180 -o ControlMaster=auto -o ControlPersist=3600s -o UserKnownHostsFile=/dev/null -o ForwardX11=no -o ProxyJump={{ __sap_vm_provision_kubevirt_vm_register_execution_host_user }}@{{ sap_vm_provision_execution_host }}"
```

Comment:
@0xFelix is there a better way in your mind to ssh into a new VM?

Reply:
Yes, by avoiding `StrictHostKeyChecking=no`. The only idea that came to my mind would be to set SSH host key via cloud init so that the fingerprint could be known. Any other idea?